File: app-android/app/src/main/java/de/tutao/tutanota/data/UserInfoDao.java
package de.tutao.tutanota.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Transaction;

import java.util.List;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public abstract class UserInfoDao {
	@Query("SELECT * FROM User")
	public abstract LiveData<List<User>> observeUsers();

	@Query("SELECT * FROM PushIdentifierKey WHERE pushIdentifierId = :pushIdentifier")
	public abstract PushIdentifierKey getPushIdentifierKey(String pushIdentifier);

	@Insert(onConflict = REPLACE)
	public abstract void insertPushIdentifierKey(PushIdentifierKey userInfo);

	@Insert(onConflict = REPLACE)
	public abstract void insertUser(User user);

	@Transaction
	public void clear() {
		clearKeys();
		clearUsers();
	}

	@Query("Delete FROM PushIdentifierKey")
	abstract void clearKeys();

	@Query("Delete FROM User")
	abstract void clearUsers();

}

File: app-android/app/src/main/java/de/tutao/tutanota/data/UserInfoDao.kt
package de.tutao.tutanota.data

import androidx.lifecycle.LiveData
import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Dao
abstract class UserInfoDao {
	@Query("SELECT * FROM User")
	abstract fun users(): List<User>

	@Query("SELECT * FROM User")
	abstract fun observeUsers(): Flow<List<User>>

	@Query("SELECT * FROM PushIdentifierKey WHERE pushIdentifierId = :pushIdentifier")
	abstract fun getPushIdentifierKey(pushIdentifier: String): PushIdentifierKey?

	@Insert(onConflict = OnConflictStrategy.REPLACE)
	abstract fun insertPushIdentifierKey(userInfo: PushIdentifierKey)

	@Insert(onConflict = OnConflictStrategy.REPLACE)
	abstract fun insertUser(user: User)

	@Transaction
	open fun clear() {
		clearKeys()
		clearUsers()
	}

	@Query("Delete FROM PushIdentifierKey")
	abstract fun clearKeys()

	@Query("Delete FROM User")
	abstract fun clearUsers()

	@Query("DELETE FROM User WHERE userId = :userId")
	abstract fun deleteUser(userId: String)
}